common/libs/units/units.h

```diff
@@ -218,18 +220,18 @@ constexpr VolumetricFlow ml_per_min(float ml_per_min) {
 //
 class Time;
 
-// Represents a length of time.
+// Represents a non-negative length of time.
```

Comment:
Duration should be allowed to be negative.  Otherwise it's super-unsafe, you subtract two times in the wrong order and you infinite loop.

Time can be a uint64 or an int64, whatever is convenient for you.